src/UI/Buyer/src/app/services/order/order-state.service.ts

```diff
@@ -128,7 +128,10 @@ export class OrderStateService {
     } else if (ordersNeverSubmitted.Items.length) {
       this.order = ordersNeverSubmitted.Items[0]
     } else if (this.appConfig.anonymousShoppingEnabled) {
-      this.order = { ID: this.tokenHelper.getAnonymousOrderID() }
+      this.order = (await Orders.Create('Outgoing', 
+      { ID: this.tokenHelper.getAnonymousOrderID(),
+        ...this.DefaultOrder as Order }
+      )) as HSOrder
```

Comment:
We actually want to defer the creating of an order until the *first* line item is added, otherwise we just create a bunch of garbage orders that pile up. See this jira task where Winmark just updated - https://four51.atlassian.net/browse/WIN-2828

Might also be worth checking out how they are doing anonymous shopping to see if there are any tricks we can steal. In particular I remember one scenario that needed to be handled is an anonymous user that added stuff to their cart and then logged in. I know we have a transfer cart mechanism during checkout but want to make sure we're accounting for that as well